lib_eio/unix/process.ml

```diff
@@ -118,8 +118,8 @@ end) = struct
   type tag = [ `Generic | `Unix ]
 
   let pipe _ ~sw =
-    (Private.pipe sw :> ([Eio.Resource.close_ty | Eio.Flow.source_ty] r *
-    [Eio.Resource.close_ty | Eio.Flow.sink_ty] r))
+    (Private.pipe sw :> ([< Eio.Resource.close_ty | Eio.Flow.source_ty] r *
```

Comment:
I don't think we need these coercions here either.

Reply:
yes yes.
this is going to be better right?
```
let pipe _ ~sw =
    let source, sink = Private.pipe sw in
    (source, sink)
```